packages/automation-contracts/scheduler/test/VestingSchedulerV2.t.sol

```diff
@@ -1696,15 +1746,101 @@ contract VestingSchedulerV2Tests is FoundrySuperfluidTester {
 
         uint32 endDate = startDate + totalDuration;
 
-        uint256 remainderAmount = totalAmount - SafeCast.toUint256(flowRate) * totalDuration;
+        uint96 remainderAmount = SafeCast.toUint96(totalAmount - SafeCast.toUint256(flowRate) * totalDuration);
 
         expectedSchedule = IVestingSchedulerV2.VestingSchedule({
             cliffAndFlowDate: cliffAndFlowDate,
             endDate: endDate,
-            claimValidityDate: 0,
             flowRate: flowRate,
             cliffAmount: cliffAmount,
-            remainderAmount: remainderAmount
+            remainderAmount: remainderAmount,
+            claimValidityDate: claimPeriod == 0 ? 0 : startDate + claimPeriod
         });
     }
+
+    function test_getMaximumNeededTokenAllowance_should_end_with_zero_if_extreme_ranges_are_used(
+        uint256 totalAmount,
+        uint32 totalDuration,
+        uint32 cliffPeriod,
+        uint32 startDate,
+        uint8 randomizer
+    ) public {
+        // Assume
+        vm.assume(randomizer != 0);
+        
+        vm.assume(startDate == 0 || startDate >= block.timestamp);
```

Comment:
generally vm.bound is recommended instead of vm.assume. check their documentation on why.